frame/nfts/src/functions.rs

```diff
@@ -179,7 +194,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				let collection_details =
 					maybe_collection_details.as_mut().ok_or(Error::<T, I>::UnknownCollection)?;
 
-				if let Ok(max_supply) = CollectionMaxSupply::<T, I>::try_get(&collection) {
+				let collection_config = Self::get_collection_config(&collection)?;
+				with_details_and_config(collection_details, &collection_config)?;
```

Comment:
I dont get the design here why this callback is needed. You can just do the checks later and rollback the changes since we have transactional layers?